### ISSUE TITLE
Some of the unpackers are misnamed

### ISSUE THREAD

**Issue by Yay295** [python-pillow/Pillow#8021]
RGB15 actually reads data as XBGR (XBBBBBGGGGGRRRRR).

https://github.com/python-pillow/Pillow/blob/a8f434f67657d9286bafb0f132ac608276fa96c0/src/libImaging/Unpack.c#L661-L674

RGBA15 actually reads data as ABGR (ABBBBBGGGGGRRRRR).

https://github.com/python-pillow/Pillow/blob/a8f434f67657d9286bafb0f132ac608276fa96c0/src/libImaging/Unpack.c#L676-L689

RGBA4B actually reads data as ABGR (AAAABBBBGGGGRRRR).

https://github.com/python-pillow/Pillow/blob/a8f434f67657d9286bafb0f132ac608276fa96c0/src/libImaging/Unpack.c#L766-L779

etc.

Basically, all of the unpackers that read less than 8 bits per band appear to be backwards.

**Comment by Yay295** [comment]
This is related to #8019, because I wrote those packers in the correct order, which is actually the opposite of the unpackers.

**Comment by Yay295** [comment]
Here's the names they should be based on what they actually do and the format specified in `Unpack.c`.

https://github.com/python-pillow/Pillow/blob/824db7152d1159bf3a895e7699b16ebc77298e77/src/libImaging/Unpack.c#L1534-L1543

RGB;15 → XBGR;1555
RGB;16 → BGR;565
BGR;5 → XRGB;1555
BGR;15 → XRGB;1555
BGR;16 → RGB;565
RGB;4B  → XBGR;4
RGBA;4B  → ABGR;4
RGBA;15 → ARGB;1555
BGRA;15 → ABGR;1555

Unfortunately the specified format doesn't quite work because the bands in these modes aren't all the same size, so I just listed all of the sizes for those modes.

**Comment by radarhere** [comment]
You're saying that when BGR;16 is decoded into RGB, it is done with the channels reversed. If that is the case, then why is https://github.com/python-pillow/Pillow/blob/main/Tests/images/bmp/g/rgb16-565pal.bmp able to be read correctly?

```python
from PIL import Image
with Image.open("Tests/images/bmp/g/rgb16-565pal.bmp") as im:
	print(im.tile)  # [('raw', (0, 0, 127, 64), 1090, ('BGR;16', 256, -1))]
	print(im.mode)  # RGB
	im.save("out.png")
```
gives

![out](https://github.com/python-pillow/Pillow/assets/3112309/4f86c6db-e1d6-41da-8979-bf224b9637d7)

just as https://entropymine.com/jason/bmpsuite/bmpsuite/html/bmpsuite.html says that it should.

**Comment by radarhere** [comment]
There's a slightly awkward overlap between this and #7965.

**Comment by Yay295** [comment]
Yes. This change would supersede both #7965 and #8019.

**Comment by Yay295** [comment]
I have [a branch for these changes](https://github.com/Yay295/Pillow/tree/16bit_rgb_rawmodes), but it's based on #8026 because it affects the same test file.

**Comment by radarhere** [comment]
That branch has now become #8158
